transformation/transformationPlugins/jsonMessageTransformers/jsonTypeMappingsSanitizationTransformer/src/main/java/org/opensearch/migrations/transform/TypeMappingsSanitizationTransformer.java

```diff
@@ -7,49 +7,58 @@
+import org.opensearch.migrations.transform.typemappings.SourceProperties;
+
 import com.google.common.io.Resources;
 
 public class TypeMappingsSanitizationTransformer extends JinjavaTransformer {
 
-    public static final String REPLAYER_VARIANT = "jinjava/typeMappings/replayer.j2";
+    public static final String ENTRYPOINT_JINJA_TEMPLATE = "jinjava/typeMappings/transformByTypeOfSourceInput.j2";
 
-    public TypeMappingsSanitizationTransformer()
+    public TypeMappingsSanitizationTransformer(
+        Map<String, Map<String, String>> indexMappings,
+        List<List<String>> regexIndexMappings)
         throws IOException {
-        this(REPLAYER_VARIANT, null, null, null);
+        this(indexMappings, regexIndexMappings, null, null, null);
     }
 
-    public TypeMappingsSanitizationTransformer(Map<String, Map<String, String>> indexMappings,
-                                               List<List<String>> regexIndexMappings)
-        throws IOException {
-        this(REPLAYER_VARIANT, null, indexMappings, regexIndexMappings);
-    }
-
-    public TypeMappingsSanitizationTransformer(String variantName,
-                                               Map<String, Object> featureFlags,
-                                               Map<String, Map<String, String>> indexMappings,
-                                               List<List<String>> regexIndexMappings) throws IOException {
+    public TypeMappingsSanitizationTransformer(
+        Map<String, Map<String, String>> indexMappings,
+        List<List<String>> regexIndexMappings,
+        SourceProperties sourceProperties,
+        Map<String, Object> featureFlags,
+        JinjavaConfig jinjavaSettings)
+        throws IOException
+    {
         super(
-            makeTemplate(variantName),
-            makeSourceWrapperFunction(featureFlags, indexMappings, regexIndexMappings));
+            makeTemplate(),
+            makeSourceWrapperFunction(sourceProperties, featureFlags, indexMappings, regexIndexMappings),
+            Optional.ofNullable(jinjavaSettings).orElse(new JinjavaConfig()));
     }
 
     private static UnaryOperator<Map<String, Object>>
-    makeSourceWrapperFunction(Map<String, Object> featureFlagsIncoming,
+    makeSourceWrapperFunction(SourceProperties sourceProperties,
+                              Map<String, Object> featureFlagsIncoming,
                               Map<String, Map<String, String>> indexMappingsIncoming,
                               List<List<String>> regexIndexMappingsIncoming)
     {
         var featureFlags = featureFlagsIncoming != null ? featureFlagsIncoming : Map.of();
         var indexMappings = indexMappingsIncoming != null ? indexMappingsIncoming : Map.of();
+        // By NOT including a backreference, we're a bit more efficient, but it also lets us be agnostic to what
+        // types of patterns are being used.
+        // This regex says, match the type part and reduce it to nothing, leave the index part untouched.
```

Comment:
Blocker

I've spent a bit trying to understand the wider implications of this default, and I'm still not groking.  Can you please expand on the comments here and explore how this default fits into the bigger picture?